session/session.go

```diff
@@ -2437,9 +2437,9 @@ func BootstrapSession(store kv.Storage) (*domain.Domain, error) {
 
 	if newCollationEnabled {
 		collate.EnableNewCollations()
-		if cfg.Experimental.EnableNewCharset {
-			collate.EnableNewCharset()
-		}
+	}
+	if cfg.Experimental.EnableNewCharset {
```

Comment:
So the default collation of GBK is still `gbk_chinese_ci`?

Reply:
yes, but it will not be respected.

Reply:
Shall we deal with it? Feels like a good idea to ban it. Besides, we can add related tests for `gbk_bin` and `gbk_chinese_ci`.

Reply:
no, there is nothing should do, just like the collation of the other charsets. if the new collation is not enabled, we can use any collation but the result does not respect the used collation.

Reply:
Get it. Just add two tests? For example, get the results of `show collation` when `EnableNewCharset` is true and `newCollationEnabled` is false.